gdx/src/com/badlogic/gdx/scenes/scene2d/ui/TextButton.java

```diff
@@ -43,7 +43,16 @@ public TextButton (String text, TextButtonStyle style) {
 		super();
 		setStyle(style);
 		this.style = style;
-		label = new Label(text, new LabelStyle(style.font, style.fontColor));
+		LabelStyle labelStyle;
+		if (style.labelStyle == null) {
+			labelStyle = new LabelStyle(style.font, style.fontColor);
+		} else {
+			labelStyle = style.labelStyle.copy();

```

Comment:
Copying the style might not be what you want. It's nice for configuring a specific TextButton without affecting others, but you may want to use the same style everywhere so a change to the style affects all buttons. By using the style without copying, copying can be done before creating the TextButton, giving the flexibility of both at the cost of some boilerplate to do the copy.